### PR TITLE
Junos testing

### PR DESCRIPTION
**Pull request by itdependsnetworks**
## Description
Update Junos testing

## Motivation and Context
Change of testing strategies.

## Impact (If any)

No longer support pytest based test for junos.

## Checklist:

- [N/A] I have updated the changelog.
- [N/A] I have updated the documentation (If applicable).
- [X] I have added tests to cover my changes (If applicable).
- [X] All new and existing tests passed.
- [X] All new code passed compilation.

### CODE REVIEW

**Comment by jeaubin** [comment]
That's great, thanks for doing this @itdependsnetworks 

We have some Junos parsers in our Dev branch right now. We can merge them here - so you can run your script on them? And all future parsers developed in house will use new structure for Junos.

**Comment by itdependsnetworks** [comment]
Can you merge mine first? I’ll pull down and create another PR for it

File: tests/ci_folder_parsing.py [hunk 1]
```diff
@@ -221,6 +255,7 @@ class FileBasedTest(aetest.Testcase):
     """Standard pyats testcase class."""
 
     OPERATING_SYSTEMS = get_operating_systems()
```
Comment (ahersi2-cisco) [inline tests/ci_folder_parsing.py:257]: in this method, I believe you should add 'junos' to the list of operating_systems that get returned if an operating systems isn't specified.

**Comment by ahersi2-cisco** [comment]
Other than the one comment above, looks good!

**Review by jeaubin — APPROVED** [review]

**Comment by jeaubin** [comment]
> Can you merge mine first? I’ll pull down and create another PR for it

Sure sounds good

**Review by ahersi2-cisco — APPROVED** [review]

**Comment by xiaoxinz-cisco** [comment]
Hi @itdependsnetworks ,

When we are developing parsers, we notice that if the parsed_output is different from the expected_output, the log doesn't show the diff between these two, only shows `AssertionError`. 

In code, it's in
```
                    # Display the diff between parsed output and golden_output
                    log.info(str(dd), extra = {'colour': 'yellow'})
                    raise AssertionError("Device output and expected output do not match")
```

Do you mind having a look at it?

Thanks

**Pull request merged**